src/tests/Microsoft.PowerFx.Core.Tests/NamedFormulasTests.cs

```diff
@@ -69,5 +70,28 @@ public void GetNamedFormulasTest(string script, string expectedX, string expecte
             Assert.Equal(expectedX, formulas.ElementAt(0).formula.Script);
             Assert.Equal(expectedY, formulas.ElementAt(1).formula.Script);
         }
+
+        [Theory]
+        [InlineData("x = 1;x = 2;", "1", 1, "x")]
+        [InlineData("y = 1;y = 2;y = 2;", "1", 2, "y")]
+        [InlineData("y = 1;y = 2;y=3", "1", 3, "y")]
+        public void NamedFormulasWithDuplicateVariablesTest(string script, string expectedX, int expectedErrorCount, string errorToken)
+        {
+            var namedFormula = new NamedFormulas(script);
+            Assert.False(namedFormula.EnsureParsed());
```

Comment:
>Assert.False(namedFormula.EnsureParsed()); [](http://example.com/codeflow?start=12&length=42)

Duplicate assigns are syntactically correct, but semantic errors. 

The "pure" way to do this would be:
- the parser succeeds, and returns a list (not dictionary)
- the binder catches the error. 

I don't think it matters now, but it might matter as we get to more IDE scenarios (getting the right error squiggles, correction scenarios)

Reply:
That's a very good point. The return here should actually be something like List<KeyValuePair<IdentToken , Formula>>, and validate this on the other side, post the correct errors on that ident token